e2e/app.e2e-spec.ts

```diff
@@ -29,27 +29,58 @@ describe('rubric-creator App', () => {
         void expect(browser.getCurrentUrl()).toContain(`/rubrics/${defaultRubricsArray[0].uuid}`);
     });
 
+    function createCustomRubric(): string {
```

Comment:
@karllhughes - Wondering your thoughts - this function is used by two tests so I extracted it, _but_ it relies on the browser already being on the `/create` page - Since this is on a spec file and we'd expect a test trying to create a custom rubric to fail if each piece isn't working, so I didn't add any sort of check into this function, but I _could_ move line 46 into this function as a catch for that . . . `void expect(browser.getCurrentUrl()).toContain('/rubrics/create');`

Reply:
I don't love the idea of having `expect` statements in a reusable function in a test.

Generally, I would only use a reusable function in a test to get to some common point and avoid duplicated code (as you've done here). Once you get into making expectations, it's probably better to stay very explicit by putting them in the test.

Thanks for asking, good question!

Reply:
Sounds good - Yeah I think this is a bit of my experience with rspec and `shared_examples` coming through - does Jasmine or Protractor have anything like that if there's multiple set-ups that get run against the same set of assertions?

Reply:
Im Jasmine, there's a `beforeEach` method that is good for putting shared logic in a group of tests: https://juristr.com/blog/2014/10/avoid-test-code-duplication-jasmine/

I'm actually not sure about protractor, but there likely is.

Reply:
Hmmmm - So rspec has before hooks as well, but I was thinking more about the [shared_examples](https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples) where you can share assertions/expectations across multiple different scenarios if those were duplicated frequently.  Not the case here, but I have found it valuable in the past. Seems like Jasmine doesn't support it directly - https://jacopretorius.net/2012/09/shared-examples-with-jasmine.html